### PR TITLE
add new b2c authentication/authorization

### PR DESCRIPTION
**Pull request by martinhenningjensen**

### CODE REVIEW

File: libraries/dotnet/EnergyOrigin.TokenValidation/EnergyOrigin.TokenValidation/B2C/B2COptions.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+using System.ComponentModel.DataAnnotations;
+
+namespace EnergyOrigin.TokenValidation.b2c;
+
+public class B2COptions
+{
+    public const string Prefix = "B2C";
+
+    [Required]
+    public string B2CWellKnownUrl { get; init; } = null!;
```
Comment (TopSwagCode) [inline libraries/dotnet/EnergyOrigin.TokenValidation/EnergyOrigin.TokenValidation/B2C/B2COptions.cs:10]: Maybe hardcode defaults for now, so we can test quicker? so we have other issue to create them in EO-BASE?

**Review by TopSwagCode — APPROVED** [review]

File: libraries/dotnet/EnergyOrigin.TokenValidation/EnergyOrigin.TokenValidation/B2C/B2COptions.cs (hunk 1 above)
Comment (martinhenningjensen) [reply libraries/dotnet/EnergyOrigin.TokenValidation/EnergyOrigin.TokenValidation/B2C/B2COptions.cs:10]: Don't we include appsettings files in deployment? (The subsystem using token validation has the configuration in appsettings)

Comment (sahma19) [reply libraries/dotnet/EnergyOrigin.TokenValidation/EnergyOrigin.TokenValidation/B2C/B2COptions.cs:10]: You will need to add environment variables to the deployment, in eo-base :)

**Review by sahma19 — CHANGES_REQUESTED** [review]
ENV variable

**Review by sahma19 — APPROVED** [review]

**Pull request merged**